lib/practice_exercises.rb

```diff
@@ -1,13 +1,44 @@
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n)
+# Space Complexity: O(1)
 def is_palindrome(string)
-  raise NotImplementedError, "Not implemented yet"
+  #check for edge case 
+  if string.length <= 1
+    return true
+  else
+    #clean up the string for comparisons
+    string = string.downcase.gsub(/[^a-z^A-Z]/, "")
+    first = 0
+    last = string.length - 1
+    until first >= last
+      #compare characters
+      return false unless string[first] == string[last]
+      first += 1
+      last -=1
+    end
+    return true
+  end
 end
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n^2)
+# Space Complexity: O(1)
 def longest_prefix(strings)
```

Comment:
I would say the time complexity is O(n * m) where m is the length of the shortest word.  Otherwise 👍